pkg/expression/builtin_cast.go

```diff
@@ -153,8 +159,11 @@ func (c *castAsIntFunctionClass) getFunction(ctx BuildContext, args []Expression
 	case types.ETString:
 		sig = &builtinCastStringAsIntSig{bf}
 		sig.setPbCode(tipb.ScalarFuncSig_CastStringAsInt)
+	case types.ETVectorFloat32:
+		sig = &builtinCastVectorFloat32AsUnsupportedSig{bf.baseBuiltinFunc}
+		// sig.setPbCode(tipb.ScalarFuncSig_CastVectorFloat32AsInt)
```

Comment:
Is there a plan to support these in the future?

Reply:
It depends. In most cases user is writing a constant vector string so that it is const-folded in the planning stage and does not need cast pushdown in the execution stage.